### PR TITLE
Customer update booking

### PR DESCRIPTION
**Pull request by phucynwa**

### CODE REVIEW

**Comment by phucynwa** [comment]
ready

**Comment by phucynwa** [comment]
ready

File: app/models/booking.rb [hunk 1]
```diff
@@ -1,11 +1,32 @@
...
+  scope :by_room, ->(room){where rooms: room}
+  scope :by_status, ->(status){where status: status}
   validates :start_time, :end_time, presence: true
+
+  def check_rooms_free
+    self.rooms.each do |room|
+      if Booking.by_room(room).by_status(:accepted).select(:start_time, :end_time)
+        .where(start_time: self.start_time..self.end_time)
+        .where(end_time: self.start_time..self.end_time)
+        .where("? BETWEEN start_time AND end_time", self.start_time).present?
```
Comment (huongnguyenmta) [inline app/models/booking.rb]: gộp dòng 18-20

**Comment by huongnguyenmta** [comment]
commented

**Comment by phucynwa** [comment]
ready

**Comment by phucynwa** [comment]
ready

**Pull request merged**